pandas/core/strings/object_array.py

```diff
@@ -0,0 +1,423 @@
+                f = lambda x: x.split(pat, n)
+            else:
+                if n is None or n == -1:
+                    n = 0
+                regex = re.compile(pat)
+                f = lambda x: regex.split(x, maxsplit=n)
+        return self._map(f, dtype=object)
+
+    def rsplit(self, pat=None, n=-1):
+        if n is None or n == 0:
+            n = -1
+        f = lambda x: x.rsplit(pat, n)
+        return self._map(f, dtype="object")
+
+    def translate(self, table):
+        return self._map(lambda x: x.translate(table))
+
+    def wrap(self, width, **kwargs):
+        kwargs["width"] = width
+        tw = textwrap.TextWrapper(**kwargs)
+        return self._map(lambda s: "\n".join(tw.wrap(s)))
+
+    def get_dummies(self, sep="|"):
+        from pandas import Series
+
+        arr = Series(self._array).fillna("")
+        try:
+            arr = sep + arr + sep
+        except TypeError:
+            arr = cast(Series, arr)
+            arr = sep + arr.astype(str) + sep
+        arr = cast(Series, arr)
+
+        tags: Set[str] = set()
+        for ts in Series(arr).str.split(sep):
+            tags.update(ts)
+        tags2 = sorted(tags - {""})
+
+        dummies = np.empty((len(arr), len(tags2)), dtype=np.int64)
+
+        for i, t in enumerate(tags2):
+            pat = sep + t + sep
+            dummies[:, i] = lib.map_infer(arr.to_numpy(), lambda x: pat in x)
+        return dummies, tags2
+
+    def upper(self):
+        return self._map(lambda x: x.upper())
+
+    def isalnum(self):
+        return self._map(str.isalnum, dtype="bool")
+
+    def isalpha(self):
+        return self._map(str.isalpha, dtype="bool")
+
+    def isdecimal(self):
+        return self._map(str.isdecimal, dtype="bool")
+
+    def isdigit(self):
+        return self._map(str.isdigit, dtype="bool")
+
+    def islower(self):
+        return self._map(str.islower, dtype="bool")
+
+    def isnumeric(self):
+        return self._map(str.isnumeric, dtype="bool")
+
+    def isspace(self):
+        return self._map(str.isspace, dtype="bool")
+
+    def istitle(self):
+        return self._map(str.istitle, dtype="bool")
+
+    def isupper(self):
+        return self._map(str.isupper, dtype="bool")
+
+    def capitalize(self):
+        return self._map(str.capitalize)
+
+    def casefold(self):
+        return self._map(str.casefold)
+
+    def title(self):
+        return self._map(str.title)
+
+    def swapcase(self):
+        return self._map(str.swapcase)
+
+    def lower(self):
+        return self._map(str.lower)
+
+    def normalize(self, form):
+        f = lambda x: unicodedata.normalize(form, x)
+        return self._map(f)
+
+    def strip(self, to_strip=None):
+        return self._map(lambda x: x.strip(to_strip))
+
+    def lstrip(self, to_strip=None):
+        return self._map(lambda x: x.lstrip(to_strip))
+
+    def rstrip(self, to_strip=None):
+        return self._map(lambda x: x.rstrip(to_strip))
+
+
+class ObjectProxy(PandasArray):
```

Comment:
Why does it need to subclass from PandasArray? Isn't only the `._str` attribute ever accessed?

Reply:
Ah, OK, the `self._array` in the ...ArrayMethods class is this object, right? Will comment on a line above then